cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaMirrorMaker2AssemblyOperator.java

```diff
@@ -415,6 +412,16 @@ private static void addClusterToMirrorMaker2ConnectorConfig(Map<String, Object>
                 jaasConfig = oauthJaasConfig.toString();
                 config.put(configPrefix + SaslConfigs.SASL_LOGIN_CALLBACK_HANDLER_CLASS, "io.strimzi.kafka.oauth.client.JaasClientOauthLoginCallbackHandler");
             }
+            else if(KafkaClientCustomAuthentication.TYPE_CUSTOM.equals(clientAuthType)) {
+                KafkaClientCustomAuthentication customAuthentication  = (KafkaClientCustomAuthentication) cluster.getAuthentication();
+                saslMechanism = customAuthentication.getSaslMechanism().isEmpty() ? "OAUTHBEARER" : customAuthentication.getSaslMechanism();
+                securityProtocol = customAuthentication.getSecurityProtocol().isEmpty() ? "SASL_SSL" : customAuthentication.getSecurityProtocol();
```

Comment:
This needs to be configured based on the `tls` configuration for the connection.